scripts/build/arch/sparc.sh

```diff
@@ -2,4 +2,17 @@
 CT_DoArchTupleValues() {
     # That's the only thing to override
     CT_TARGET_ARCH="sparc${target_bits_64}${CT_ARCH_SUFFIX}"
+
+    # By default, sparc64-*-linux is configured with -mcpu=v9. However,
+    # according to https://sourceware.org/ml/libc-alpha/2005-12/msg00027.html,
+    # "There is no Linux sparc64 port that runs on non-UltraSPARC-I+ ISA CPUs."
+    # There is a patch that would change the default to -mcpu=ultrasparc for
+    # sparc64-*-linux configuration: https://patchwork.ozlabs.org/patch/409424/
+    # but that patch has not been integrated (yet). One concern raised about
+    # this patch was that -mcpu=ultrasparc can suboptimally schedule instructions
+    # for newer SPARC CPUs. So, override to -mcpu=ultrasparc and warn the user.
```

Comment:
Yes, your understanding is correct. As to ultrasparc3, I am not a SPARC expert (i.e., does ultrasparc3 allow for some instructions that ultrasparc does not?) and I don't have the hardware to test either - so I went with the more conservative option.